### PR TITLE
Fix PostgreSQLSource reading all unread the data in onFinish

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix PostgreSQL reading all the data even though `LIMIT n` could be specified.

### CODE REVIEW

File: src/Processors/Sources/PostgreSQLSource.cpp [hunk 1]
```diff
@@ -176,12 +176,10 @@ template<typename T>
 void PostgreSQLSource<T>::onFinish()
 {
     if (stream)
-    {
-        stream->complete();
```
Comment (kssenii) [inline src/Processors/Sources/PostgreSQLSource.cpp:180]: turns out the `complete()` method read all unread data with `ignore`

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 07eedc8ef18c829b884f4fbdd579ed00822a30a8 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50187/07eedc8ef18c829b884f4fbdd579ed00822a30a8/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by vdimir — APPROVED** [review]

**Comment by kssenii** [comment]
> Stateless tests (release, wide parts enabled) — fail: 1

`02417_opentelemetry_insert_on_distributed_table` - not related to changes

> Stateless tests (debug, s3 storage) [6/6] — Tests are not finished

https://github.com/ClickHouse/ClickHouse/issues/49185

**Comment by helifu** [comment]
Hi @kssenii, could you please share why the value here is 18?
https://github.com/ClickHouse/ClickHouse/blob/53de761e0eb6c234ff1f2f7d0afb653d31b3ebbe/tests/integration/test_storage_postgresql/test.py#LL341C8-L341C8

In addition, there was an exception in my PR:
https://s3.amazonaws.com/clickhouse-test-reports/49778/e9ffc383da0e544cc01bbad7717dae3f566e95f5/integration_tests__asan__[4_6]/integration_run_parallel2_0.log

Thanks in advance :)

**Pull request merged**